crates/net/network/src/transactions.rs

```diff
@@ -305,9 +306,12 @@ where
         // number of connected peers)
         let max_num_full = (self.peers.len() as f64).sqrt() as usize + 1;
 
-        // Note: Assuming ~random~ order due to random state of the peers map hasher
-        for (peer_idx, (peer_id, peer)) in self.peers.iter_mut().enumerate() {
+        // choose random peers at every propagation
+        let mut peer_ids: Vec<PeerId> = self.peers.keys().cloned().collect();
+        peer_ids.shuffle(&mut rand::thread_rng());
```

Comment:
this is the obvious solution, but I put this off because this is very expensive, considering how often this is invoked. This now does a clone, shuffle, and additional lookups all are more expensive with more peers

this could be designed better in such way a that it doesn't require a new allocation and expensive shuffleing, even pseudo-random would be fine here I believe.
I guess a short-term fix would be to reshuffle the map on some interval and do that when we have the time (when task is pending)

also, I don't think geth even shuffles peers everytime

https://github.com/ethereum/go-ethereum/blob/0004c6b229b787281760b14fb9460ffd9c2496f1/eth/handler.go#L566-L566